### PR TITLE
Allow a maximum of 64 bits to be returned in registers

### PR DESCRIPTION
**Pull request by shepmaster**
The rest spills to the stack

### CODE REVIEW

File: lib/Target/AVR/AVRISelLowering.cpp [hunk 1]
```diff
@@ -1295,6 +1295,20 @@ SDValue AVRTargetLowering::LowerCallResult(
...
+AVRTargetLowering::CanLowerReturn(CallingConv::ID CallConv,
+                                  MachineFunction &MF, bool isVarArg,
+                                  const SmallVectorImpl<ISD::OutputArg> &Outs,
+                                  LLVMContext &Context) const
+{
+  unsigned TotalBits = 0;
+  for (auto Out : Outs) {
+    TotalBits += Out.VT.getSizeInBits();
+  }
+  unsigned MaxBits = 64;
```
Comment (dylanmckay) [inline lib/Target/AVR/AVRISelLowering.cpp]: Add a small comment about why `64` is the max, and that the other data is spilled.

**Comment by dylanmckay** [comment]
LGTM with minor comment nit.

**Comment by dylanmckay** [comment]
Have also added you as a collaborator, you should be able to merge this yourself now

File: lib/Target/AVR/AVRISelLowering.cpp (hunk 1 above)
Comment (shepmaster) [inline lib/Target/AVR/AVRISelLowering.cpp]: Ideally, I'd love to be able to ask the calling convention about this information; do you know if there's some way of doing that introspection? I'd hope that would remove the need for the comment.

Comment (dylanmckay) [inline lib/Target/AVR/AVRISelLowering.cpp]: Nah, I don't believe there is. I don't like the way CC handling is implemented in LLVM, it's not very flexible, and it definitely is ugly.

**Comment by shepmaster** [comment]
@dylanmckay I reworked this to make it a bit more generic, after re-reading and understanding the ARM implementation a bit more.

**Comment by dylanmckay** [comment]
Looks good to me!

**Pull request merged**